webscraper/dataScraperAPI.js

```diff
@@ -10,65 +10,221 @@ function scrape() {
-    for (let i = 3; i < histWindTbodyLen + 1; i++) {
-      // only interested in times with :40 to sync up with histSwellInfo
-      if (regex.test($("table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(" + i + ") > td:nth-child(3)").text().trim())) {
-        histWindInfo[j] = {
-          'month'         : $("table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(" + i + ") > td:nth-child(1)").text().trim(),
-          'day'           : $("table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(" + i + ") > td:nth-child(2)").text().trim(),
-          'time'          : $("table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(" + i + ") > td:nth-child(3)").text().trim(),
-          "windDirection" : $('table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(4)').text().trim(),
-          "windSpeed"     : $('table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' + i + ') > td:nth-child(5)').text().trim(),
-          }
+      // Scraping info for histWindInfo
+      let histWindTbodyLen = $('table.dataTable:nth-child(4) > tbody:nth-child(2)').children(
+        'tr',
+      ).length;
+      let j = 0;
+      let regex = new RegExp(':40');
+      for (let i = 3; i < histWindTbodyLen + 1; i++) {
+        // only interested in times with :40 to sync up with histSwellInfo
+        if (
+          regex.test(
+            $(
+              'table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' +
+                i +
+                ') > td:nth-child(3)',
+            )
+              .text()
+              .trim(),
+          )
+        ) {
+          histWindInfo[j] = {
+            month: $(
+              'table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' +
+                i +
+                ') > td:nth-child(1)',
+            )
+              .text()
+              .trim(),
+            day: $(
+              'table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' +
+                i +
+                ') > td:nth-child(2)',
+            )
+              .text()
+              .trim(),
+            time: $(
+              'table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' +
+                i +
+                ') > td:nth-child(3)',
+            )
+              .text()
+              .trim(),
+            windDirection: $(
+              'table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' +
+                i +
+                ') > td:nth-child(4)',
+            )
+              .text()
+              .trim(),
+            windSpeed: $(
+              'table.dataTable:nth-child(4) > tbody:nth-child(2) > tr:nth-child(' +
+                i +
+                ') > td:nth-child(5)',
+            )
+              .text()
+              .trim(),
+          };
           j++;
         }
-      else 
         // do nothing
-        continue;
+        else continue;
```

Comment:
Why did you choose to use `else continue` here? Are you purposefully trying to step out of the current iterative block in your loop and move to the next iteration?

---

_very subjective_

Note that some may think that usage of `continue`  in this context is an anti-pattern. I'm not experienced enough to know for sure. However, I do think that it's unnecessary in the given code block. I think refactoring the logic and ordering with `if-else if-else` blocks would be easier to read (and what's easier to read is easier to maintain).

Took me a while to appreciate that code is for humans rather than machines. Keeping that perspective in mind helped me a lot.